### PR TITLE
EDUCATOR-1817 | be defensive about serializing training examples, log…

### PR DESCRIPTION
**Pull request by iloveagent57**
… an exception when it occurs.

https://openedx.atlassian.net/browse/EDUCATOR-1817

As I mentioned in the ticket, there's a training example that's actually a list when we're expecting a dict, and it's causing a course export to consistently fail.  This PR aims to be a little more defensive about the types of data structures we might see (as we are elsewhere in ORA) and logs an exception if one occurs in the problematic area of code.

### CODE REVIEW

File: openassessment/xblock/xml.py [hunk 1]
```diff
@@ -606,9 +610,21 @@ def serialize_training_examples(examples, assessment_el):
 
         # Answer provided in the example (default to empty string)
         answer_el = etree.SubElement(example_el, 'answer')
-        for part in example_dict.get('answer', {}).get('parts', []):
-            part_el = etree.SubElement(answer_el, 'part')
-            part_el.text = unicode(part.get('text', u''))
+        try:
+            answer = example_dict.get('answer')
```
Comment (awaisdar001) [inline openassessment/xblock/xml.py]: @iloveagent57 We are initiating `parts` var here, Why we are not using it anywhere?

File: openassessment/xblock/xml.py [hunk 2]
```diff
@@ -606,9 +610,21 @@ def serialize_training_examples(examples, assessment_el):
...
+            if answer is None:
+                parts = []
+            elif isinstance(answer, dict):
+                parts = answer.get('parts', [])
+            elif isinstance(answer, list):
+                parse = answer
```
Comment (awaisdar001) [inline openassessment/xblock/xml.py]: Did you mean `parts` here?

File: openassessment/xblock/xml.py [hunk 3]
```diff
@@ -606,9 +610,21 @@ def serialize_training_examples(examples, assessment_el):
...
+
+            for part in example_dict.get('answer', {}).get('parts', []):
```
Comment (awaisdar001) [inline openassessment/xblock/xml.py]: Is this PR is just for logging? because `example_dict.get('answer', {})` will break again if it returns a `list`

Comment (iloveagent57) [reply openassessment/xblock/xml.py]: ah, thank you, that should be `for part in parts:`

File: openassessment/xblock/xml.py (hunk 2 above)
Comment (iloveagent57) [reply openassessment/xblock/xml.py]: geez, yup!

**Pull request merged**